src/main/network/dns.rs

```diff
@@ -30,27 +30,55 @@ struct Record {
 
 #[derive(Debug, PartialEq)]
 pub enum RegistrationError {
-    InvalidAddr,
-    InvalidName,
-    AddrExists,
-    NameExists,
+    BroadcastAddrInvalid,
+    LocalhostAddrInvalid,
```

Comment:
(Optional) Maybe `LoopbackAddrInvalid` instead? Localhost is usually a single IP address (typically 127.0.0.1 or whatever is in the hosts file), whereas loopback is the IP range 127.0.0.0/8.

Reply:
Ahh, thanks for noticing that. Fixed now! (in 69b5a5511c77f689f2f769ffdf65330a518a5bfe)